plugins/dell/fu-dell-common.c

```diff
@@ -60,17 +60,48 @@ _dell_smi_obj_free (FuDellSmiObj *obj)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (FuDellSmiObj, _dell_smi_obj_free);
 
 gboolean
-fu_dell_supported (void)
+fu_dell_supported (GError **error)
 {
 	guint8 dell_supported = 0;
 	struct smbios_struct *de_table;
+	g_autofree gchar *dmidecode_path = NULL;
+	g_autofree gchar *dmidecode_version = NULL;
+	g_autofree gchar *oem_string = NULL;
+	gboolean ret;
+
+	/* hwids are not reliable enough to detect this due to potential OEM rebranding
+	 * and systems that are branded but don't support all SMBIOS extensions
+	 */
+	dmidecode_path = g_find_program_in_path ("dmidecode");
+	if (dmidecode_path != NULL) {
+		ret = g_spawn_command_line_sync ("dmidecode --v",
```

Comment:
Does `fu_common_spawn_sync` help you here? I'm assuming we can't get the output of ` --oem-string 1` from sysfs.

Reply:
Wasn't aware of that helper function.  I'll check on it instead.
And yeah, nothing in sysfs (see previous comment).

Reply:
That seems like overkill to me since the output i'm fetching is always 1 line.

Reply:
True.